utensor_cgen/frontend/tensorflow.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+import tensorflow as tf
+import numpy as np
+from google.protobuf import text_format
+
+from utensor_cgen.frontend.base import Parser
+from utensor_cgen.frontend import FrontendSelector
+from utensor_cgen.ir.base import TensorInfo, OperationInfo, uTensorGraph
+from utensor_cgen.utils import topologic_order_graph
+
+
+@FrontendSelector.register(target_exts=['.pb', '.pbtxt'])
```

Comment:
I'm not a fan of selecting the frontend parsers based on file extensions. If anything, we need to load from protobuf and determine what kind of data structure, ugraph, custom_graph, etc. is described by the file and proceed from there.

Reply:
So do you have any suggestion how to determine which frontend to use other than file extension?

Reply:
As far as what I know, there is no common way to detect the content format.
Something like: if the first few bytes is blah then TF, if blah then onnx else blah blah blah.
So file extension is the simplest way I can think of.

Reply:
May be just reading the first "node" in the pb file. If it is a graph_def it should just be a list of NodeDefs. As long as the first entry is a NodeDef we can assume its a graph def format.

Alternatively we can be super lazy:
```
try:
  graph_def_import()
except GraphDefImportExceptionNotGraphDef:
  logger.info("Whoops not a graph def")
try:
  import_torch_from_pb()
except ...
...
```
XD

Reply:
Honestly, I don't think it's a good solution.
As the community is concerned, the file extension convention is very well-known.

The intension of my decorator api is very explicit.
(Unfortunately I have to support python2 or I'll make `target_exts` a keyword-only argument)
So I think it would be easy for future developers/users to adapt to this approach.

How do you think? @Knight-X @neil-tan @mbartling

Reply:
I think this is a case of a more robust error handling. The extension should be enough for us to select the frontend in the most cases. @mbartling's case should result in exceptions later in the parses. We can choose to either warn the users or fool-proof the error. We can do that in future improvement PRs perhaps?

Reply:
Sure, we can do that.
I think we can do it with a cli option such as:
```
# by the file extension
utensor-cli convert whatever_graph --output-nodes=out1,out2 --frontend-parse=default
# for trying every possible parser and use the first match
utensor-cli convert whatever_graph --output-nodes=out1,out2 --frontend-parse=all
# or force to use specific parser
utensor-cli convert whatever_graph --output-nodes=out1,out2 --frontend-parse=tensorflow
```
just name a few.

I'll add it to issue tracker